File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        int n = deck.length;
        
        Deque<Integer> dqe = new ArrayDeque<>();
        Arrays.sort(deck);

        for (int i = n - 1; i > -1; i--) {
            if (!dqe.isEmpty())
                dqe.addFirst(dqe.removeLast());

            dqe.addFirst(deck[i]);
        }

        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = dqe.removeFirst();

        return arr;
    }
}

File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.js
var deckRevealedIncreasing = function (deck) {
    const n = deck.length;
    deck.sort((a, b) => a - b);

    const dqe = [];

    for (let i = n - 1; i > -1; i--) {
        if (dqe.length > 0)
            dqe.unshift(dqe.pop());

        dqe.unshift(deck[i]);
    }

    const arr = new Array(n);
    for (let i = 0; i < n; i++)
    arr[i] = dqe.shift();

    return arr;
};